new_core/filelib.js

```diff
@@ -0,0 +1,94 @@
+/* Json File Library
+ *
+ * A library for handling files with helper function for JSON type files
+ */
+
+const json = {
+	request: undefined,
+	busy: 0,
+	data: [],
+	queue: []
+}
+
+export function addFile(fileurl) {
+
+	// This will add things to the queue if things get busy
+	if(json.busy == 1){
+		json.queue.push(fileurl);
+		return;
+	}
+
+	json.busy = 1
+
+	// This will get the right object for XMLHttpRequest
+	if(window.XMLHttpRequest){
+		alert("Gets here")
```

Comment:
can you replace this by console.log?

Reply:
Yeah, I think a bit of cleaning up alerts is in order here